client/lib/sites-list/list.js

```diff
@@ -464,11 +465,16 @@ SitesList.prototype.getSelectedOrAllJetpackCanManage = function() {
 };
 
 SitesList.prototype.getSelectedOrAllWithPlugins = function() {
-	return this.getSelectedOrAll().concat(
+	return unique( this.getSelectedOrAll().concat(
 		this.getSelectedOrAll().filter( function( site ) {
 			return isBusiness( site.plan );
-		} )
-	);
+		} ) ), 'ID' );
+};
+
+SitesList.prototype.getSelectedOrWPComBusiness = function() {
+	return this.getSelectedOrAll().filter( function( site ) {
+		return isBusiness( site.plan );
+	} );
```

Comment:
For kicks and giggles, here is another way I might style this function which suits my personality closer - YMMV

```js
const isBusinessSitePlan = site => isBusiness( site.plan );

SitesList.prototype.getSelectedOrWpComBusiness = () => (
	this
		.getSelectedOrAll()
		.filter( isBusinessSitePlan );
);
```